### PR TITLE
Handle new TS Types for doc generation

### LINKED ISSUE #305: `ui-extensions` use new version of `docs-tools` and output new types
When this PR https://github.com/Shopify/docs-tools/pull/15 is merged and a new package is published, use the new package and output the new types it supports in the appropriate format.

### PR DESCRIPTION
**Pull request by jamesvidler**
### Background
Closes #305 and https://github.com/Shopify/docs-tools/issues/18

This PR implements the new `docs-tools` package done in https://github.com/Shopify/docs-tools/pull/15. It adds some additional logic to appropriately handle these new types and output them accordingly.

### Solution
- Implement new (unhandled TS types) in https://github.com/Shopify/docs-tools/pull/15
- Use the new package (once published)
- Add some logic to ensure these new types are outputted correctly in the docs

### 🎩

See tophatting instructions from sibling PR https://github.com/Shopify/docs-tools/pull/15

### Checklist

- [x] I have :tophat:'d these changes

### CODE REVIEW

File: scripts/typedoc/shopify-dev-renderer/shared/index.ts [hunk 1]
```diff
@@ -186,6 +186,8 @@ function propType(
   const PIPE = '&#124;';
 
   switch (value.kind) {
+    case 'UndefinedType':
```
Comment (jamesvidler) [inline scripts/typedoc/shopify-dev-renderer/shared/index.ts:189]: Resolves the issue renaming UndefinedType to `undefined`.

File: scripts/typedoc/shopify-dev-renderer/shared/index.ts [hunk 2]
```diff
@@ -209,17 +211,20 @@ function propType(
...
     case 'Local':
       // eslint-disable-next-line no-case-declarations
       const local = exports.find(
         ({value: exportValue}: any) => exportValue.name === value.name,
       );
 
       if (local == null) {
-        console.warn(
-          `Can’t resolve export type \`${value.name}\` in ${dir}. Maybe it’s not exported from the component index or imported from a remote package.`,
-        );
```
Comment (jamesvidler) [inline scripts/typedoc/shopify-dev-renderer/shared/index.ts:221]: This was outputting way too much noise that didn't need to be actioned on.

File: scripts/typedoc/shopify-dev-renderer/shared/index.ts [hunk 3]
```diff
@@ -209,17 +211,20 @@ function propType(
...
 
+        if (value.name === 'Unknown' && value?.params[0]?.name) {
+          // HACK: to get the TSMethodSignature name to show instead of simply "Unknown"
```
Comment (jamesvidler) [inline scripts/typedoc/shopify-dev-renderer/shared/index.ts:225]: This looks a little too hacky... must be a safer way to do this.

**Comment by rennyG** [comment]
@jamesvidler peep the [dev docs/tophat PR](https://github.com/Shopify/shopify-dev/pull/19922). `yarn docs:checkout` outputted some unhandled TS type messages

**Comment by rennyG** [comment]
@jamesvidler second step in the linked tophat instructions say

`Run yarn link @shopify/ui-extensions in ui-extensions`

that outputs this for me 

```
renchaturvedi@Rens-MBP ui-extensions % yarn link @shopify/ui-extensions
yarn link v1.19.2
error No registered package found called "@shopify/ui-extensions".
```

Noticed on `docs-tools` we're using a different version, not sure if that's my issue?

```
renchaturvedi@Rens-MBP docs-tools % yarn link
yarn link v1.22.5
success Registered "@shopify/docs-tools".
info You can now run `yarn link "@shopify/docs-tools"` in the projects where you want to use this package and it will be used instead.
✨  Done in 0.12s.
```

**Review by rennyG — APPROVED** [review]
Updated my PR using the tophat instructions in https://github.com/Shopify/docs-tools/pull/15 and it looks good!

https://github.com/Shopify/shopify-dev/pull/19922

**Pull request merged**